core/src/test/java/org/elasticsearch/aliases/IndexAliasesIT.java

```diff
@@ -427,6 +428,23 @@ public void testDeleteAliases() throws Exception {
 
         AliasesExistResponse response = admin().indices().prepareAliasesExist(aliases).get();
         assertThat(response.exists(), equalTo(false));
+
+        logger.info("--> creating index [foo_foo] and [bar_bar]");
+        assertAcked(prepareCreate("foo_foo"));
+        assertAcked(prepareCreate("bar_bar"));
+        ensureGreen();
+
+        logger.info("--> adding [foo] alias to [foo_foo] and [bar_bar]");
+        assertAcked(admin().indices().prepareAliases().addAlias("foo_foo", "foo"));
+        assertAcked(admin().indices().prepareAliases().addAlias("bar_bar", "foo"));
+
+        assertAcked(admin().indices().prepareAliases().addAliasAction(AliasActions.remove().index("foo*").alias("foo")).execute().get());
+
+        assertThat(admin().indices().prepareAliasesExist("foo").get().exists(), equalTo(true));
+        assertThat(admin().indices().prepareAliasesExist("foo").setIndices("foo_foo").get().exists(), equalTo(false));
+        assertThat(admin().indices().prepareAliasesExist("foo").setIndices("bar_bar").get().exists(), equalTo(true));
```

Comment:
oh man, I used to agree with you till somebody somehow convinced me that we should do it with assertEquals because it's simpler. I am sure there are comments in the github history around this. I really do not mind either way after all. Next time I will leave things as they are.